packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +304,637 @@ class Cubic extends Curve {
   }
 }
 
+/// Abstract class that defines an API for evaluating 2D parametric curves.
+///
+/// [Curve2D] differs from [Curve] in that the values interpolated are [Offset]
+/// values instead of [double] values, hence the "2D" in the name. They both
+/// take a single double `t` that has a range of 0.0 to 1.0, inclusive, as input
+/// to the [transform] function . Unlike [Curve], [Curve2D] is not required to
+/// map `t=0.0` and `t=1.0` to specific output values.
+///
+/// The interpolated `t` value given to [transform] represents the progression
+/// along the curve, but it doesn't necessarily progress at a constant velocity, so
+/// incrementing `t` by, say, 0.1 might move along the curve by quite a lot at one
+/// part of the curve, or hardly at all in another part of the curve, depending
+/// on the definition of the curve.
+abstract class Curve2D extends ParametricCurve<Offset> {
+  /// Abstract const constructor to enable subclasses to provide const
+  /// constructors so that they can be used in const expressions.
+  const Curve2D();
+
+  /// Generates a list of samples with a recursive subdivision until a tolerance
+  /// of `tolerance` is reached.
+  ///
+  /// Samples are generated in order.
+  ///
+  /// Samples can be used to render a curve efficiently, since the samples
+  /// constitute line segments which vary in size with the curvature of the
+  /// curve. They can also be used to quickly approximate the value of the curve
+  /// by searching for the desired range in X and linearly interpolating between
+  /// samples to obtain an approximation of Y at the desired X value. The
+  /// implementation of [CatmullRomCurve] uses samples for this purpose
+  /// internally.
+  ///
+  /// The tolerance is computed as the area of a triangle formed by a new point
+  /// and the preceding and following point.
+  ///
+  /// See also:
+  ///
+  ///  * Luiz Henrique de Figueire's Graphics Gem on [the algorithm](http://ariel.chronotext.org/dd/defigueiredo93adaptive.pdf).
+  Iterable<Curve2DSample> generateSamples({double start = 0.0, double end = 1.0, double tolerance = 1e-10}) {
```

Comment:
Typically we insert spaces around named parameter lists (here and elsewhere).
```dart
({ double start = 0.0, double end = 1.0, double tolerance = 1e-10 }) 
```

Reply:
Actually, it's pretty long, so I just made it multi-line.